### PR TITLE
Serize

### PR DESCRIPTION
**Pull request by joseserize0222**

### CODE REVIEW

File: Main.hs [hunk 1]
```diff
@@ -1,25 +1,59 @@
...
 
 import Control.Monad (unless)
 import Text.Printf (printf)
 
 short :: [a] -> Bool
-short = undefined 
+short [] = True
+short [x] = True
+short [x, y] = True
+short xs = False
```
Comment (kajigor) [inline Main.hs:12]: There is a way to implement this function in 2 lines. Can you do it? 

Also, avoid introducing variables names you aren't using. Instead, use wildcards.

File: Main.hs [hunk 2]
```diff
@@ -1,25 +1,59 @@
...
 
 lovely :: [Int] -> Bool
-lovely = undefined 
+lovely [_, _ , 14, _] = True
+lovely xs = short xs
```
Comment (kajigor) [inline Main.hs:15]: What about `[1,2,14]`?

File: Main.hs [hunk 3]
```diff
@@ -1,25 +1,59 @@
...
 
 isLeapYear :: Int -> Bool
-isLeapYear year = undefined 
+isLeapYear year = case year `mod` 4 of
+  0 -> case year `mod` 100 of
+              0 -> case year `mod` 400 of
+                        0 -> True
+                        _ -> False
+              _ -> True
+  _ -> False
```
Comment (kajigor) [inline Main.hs:55]: What about negative years?

**Review by kajigor — CHANGES_REQUESTED** [review]

File: Main.hs [hunk 4]
```diff
@@ -1,25 +1,58 @@
+{-# OPTIONS_GHC -Wno-overlapping-patterns #-}
 module Main where
 
 import Control.Monad (unless)
 import Text.Printf (printf)
 
 short :: [a] -> Bool
-short = undefined 
+short xs = length (take 3 xs) <= 2
```
Comment (alex28sh) [inline Main.hs:8]: There is an overhead in your solution

File: Main.hs [hunk 5]
```diff
@@ -1,25 +1,58 @@
...
 
 isLeapYear :: Int -> Bool
-isLeapYear year = undefined 
+isLeapYear year = if year >= 0
+  then
+    case year `mod` 4 of
+      0 -> case year `mod` 100 of
+                  0 -> case year `mod` 400 of
+                            0 -> True
+                            _ -> False
```
Comment (alex28sh) [inline Main.hs:51]: it seems, like your solution has too much nesting

I would advice to avoid writing such code

**Comment by alex28sh** [comment]
ok

**Pull request closed without merging**